### PR TITLE
CORE-19316: log at info on first time

### PR DESCRIPTION
**Pull request by dickon**
only log healthy on upward transitions

### CODE REVIEW

File: applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/Health.kt [hunk 1]
```diff
@@ -43,12 +38,12 @@ object Health {
         val statusRouteHandler = WebHandler { context ->
             val notReadyComponents = lifecycleRegistry.componentWithStatus(setOf(LifecycleStatus.DOWN, LifecycleStatus.ERROR))
             val status = if (notReadyComponents.isEmpty()) {
```
Comment (davidcurrie) [inline applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/Health.kt]: Do we also want to add logging to the ready state i.e. `All components are up`?

File: applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/Health.kt [hunk 2]
```diff
@@ -26,15 +26,10 @@ object Health {
                 if (healthy)
                     "Status is healthy"
```
Comment (davidcurrie) [inline applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/Health.kt]: Given @kyriathar comments, is it worth changing this to be something more explicit e.g. `Status is healthy. No components have errors.`?

File: applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/Health.kt [hunk 3]
```diff
@@ -60,14 +55,17 @@ object Health {
         webServer.registerEndpoint(Endpoint(HTTPMethod.GET, HTTP_STATUS_ROUTE, statusRouteHandler))
     }
 
-    private fun clearLastLogMessageForRoute(route: String) {
-        lastLogMessage[route] = ""
-    }
 
-    private fun logIfDifferentFromLastMessage(route: String, logMessage: String) {
+
+    private fun logIfDifferentFromLastMessage(route: String, logMessage: String, healthy: Boolean) {
```
Comment (davidcurrie) [inline applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/Health.kt]: `healthy` is a bit misleading for the last parameter, given it's also used for `ready`

File: applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/Health.kt (hunk 2 above)
Comment (dickon) [reply applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/Health.kt]: done

File: applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/Health.kt (hunk 1 above)
Comment (dickon) [reply applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/Health.kt]: good idea, done

Comment (dickon) [reply applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/Health.kt]: now called newStatus, done

File: applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/Health.kt (hunk 3 above)
Comment (dickon) [reply applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/Health.kt]: now called newStatus, done.

**Review by davidcurrie — APPROVED** [review]
LGTM

**Pull request merged**